flags/src/main/java/com/yahoo/vespa/flags/json/Condition.java

```diff
@@ -1,62 +1,72 @@
-// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
 package com.yahoo.vespa.flags.json;
 
 import com.yahoo.vespa.flags.FetchVector;
 import com.yahoo.vespa.flags.json.wire.WireCondition;
 
-import java.util.Arrays;
 import java.util.List;
 import java.util.Objects;
+import java.util.Optional;
 import java.util.function.Predicate;
 
 /**
  * @author hakonhall
  */
-public class Condition implements Predicate<FetchVector> {
-    public enum Type { WHITELIST, BLACKLIST }
+public interface Condition extends Predicate<FetchVector> {
+    enum Type {
+        WHITELIST,
+        BLACKLIST,
+        RELATIONAL;
 
-    private final Type type;
-    private final FetchVector.Dimension dimension;
-    private final List<String> values;
+        public String toWire() { return name().toLowerCase(); }
 
-    public Condition(Type type, FetchVector.Dimension dimension, String... values) {
-        this(type, dimension, Arrays.asList(values));
-    }
+        public static Type fromWire(String typeString) {
+            for (Type type : values()) {
+                if (type.name().toLowerCase().equals(typeString.toLowerCase())) {
+                    return type;
+                }
+            }
 
-    public Condition(Type type, FetchVector.Dimension dimension, List<String> values) {
-        this.type = type;
-        this.dimension = dimension;
-        this.values = values;
+            throw new IllegalArgumentException("Unknown type: '" + typeString + "'");
+        }
     }
 
-    @Override
-    public boolean test(FetchVector vector) {
-        boolean isMember = vector.getValue(dimension).filter(values::contains).isPresent();
+    class CreateParams {
+        private final FetchVector.Dimension dimension;
+        private final List<String> values;
+        private final Optional<String> predicate;
 
-        switch (type) {
-            case WHITELIST: return isMember;
-            case BLACKLIST: return !isMember;
-            default: throw new IllegalArgumentException("Unknown type " + type);
+        public CreateParams(FetchVector.Dimension dimension, List<String> values, Optional<String> predicate) {
```

Comment:
Error when setting a predicate for whitelist/blacklist or setting values for relational?

Reply:
Done